### DISCUSSION TITLE
share the internal api externally (for example via rest)

### DISCUSSION THREAD

**Discussion by asvdvl** [rigwild/discord-self-bot-console#68] [Q&A] [answered]
my goal is to implement scheduled messages, while the contents of the message need to be taken from a file on the system, for this I already have a bash script that, while my computer is not busy, inserts text into the input field and presses enter
My current solution has problems that I want to solve using this library (I liked that the token does not go outside the discord client):
- track that the message was accurately sent
   - problems with the network and discord closes the input field with its annoying loading
   - when leaving, I may forget to point to the right channel or leave the discord out of focus
- send messages while I'm using my PC

my idea is to use this library to open a server that will proxy requests from my script, how possible is this?(I also have to say that I have no experience with js or electon)

**Comment by rigwild** [comment]
This is out of topic of the repo.

Anyway, I do not think it is possible. The Discord client is running Electron which is basically a Chromium browser. The Discord client is thus a sandbox as you can only access the UI part of the browser. You cannot interact with the file system or listen to ports in that context.

This lib is mostly made for quick scripts to run very simple automations. If you which to do more than this, I suggest finding another library that would note live in the Discord client, and that accepts self bot usage, to achieve what you want.

**Comment by Necione** [comment, marked as answer]
> my goal is to implement scheduled messages

Have you tried using [Puppeteer](https://pptr.dev/) to inject a script? You can have a headless chrome window login to Discord, inject the API and then run any js script that you could use to automate your tasks.
```
  const apiInjectionScript = await fs.readFile('index.js', 'utf8'); // where index.js is the API script
  await page.evaluate(apiInjectionScript);
```
and then just 
```
  const taskScript = await fs.readFile('somescript.js', 'utf8');
  await page.evaluate(taskScript);
```
This way you can have the script running on a VPS or even your own computer, and the window doesn't even need to be open since it's running on headless Chromium. You could have it setup to run multiple scripts at once if you'd like too.

**Reply by asvdvl** [reply]
Thanks for the tip, I'll try it when I have some free time.

**Reply by rigwild** [reply]
That is very smart! Thanks for answering 🙏
